src/basic/strverscmp.c

```diff
@@ -1,20 +1,75 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
```

Comment:
oh, yikes, sharing code between the uefi codebase and the rest. so far we didn't dare and just copied stuff...

I figure doing this sooner or later is unavoidable, but i wonder if maybe we should try to isolate this code in a separate dir, so that we don't have src/basic/ where some stuff is userspace-only, other stuff also works in uefi.

i.e. maybe add a new dir maybe called `src/compatible/` or `śrc/extricable/` or `src/common-firmware/` or `src/firmware-shared/` or so? other ideas? slight preference for `src/firmware/common` from my side

The idea would be that everything in `src/boot/efi/` would be pure efi code, everything in `src/basic/` would be pure userspace code, and the stuff in `src/firmware-common/` would be the common code that works in both environments?

Most likely a bunch of stuff from src/boot/efi/util.h could be moved there too